nemo/collections/tts/models/fastpitch_hifigan_e2e.py

```diff
@@ -0,0 +1,415 @@
+        if isinstance(cfg, dict):
+            cfg = OmegaConf.create(cfg)
+        elif not isinstance(cfg, DictConfig):
+            raise ValueError(f"cfg was type: {type(cfg)}. Expected either a dict or a DictConfig")
+        # Ensure passed cfg is compliant with schema
+        OmegaConf.merge(cfg, schema)
+
+        # torch.backends.cudnn.benchmark = True  # TODO
+        self.preprocessor = instantiate(self._cfg.preprocessor)
+        self.melspec_fn = instantiate(self._cfg.preprocessor, highfreq=None, use_grads=True)
+
+        self.encoder = instantiate(self._cfg.input_fft)
+        self.duration_predictor = instantiate(self._cfg.duration_predictor)
+        self.pitch_predictor = instantiate(self._cfg.pitch_predictor)
+
+        self.generator = instantiate(self._cfg.generator)
+        self.multiperioddisc = MultiPeriodDiscriminator()
+        self.multiscaledisc = MultiScaleDiscriminator()
+        self.mel_val_loss = L1MelLoss()
+        self.feat_matching_loss = FeatureMatchingLoss()
+        self.disc_loss = DiscriminatorLoss()
+        self.gen_loss = GeneratorLoss()
+
+        self.max_token_duration = cfg.max_token_duration
+
+        self.pitch_emb = torch.nn.Conv1d(
+            1,
+            cfg.symbols_embedding_dim,
+            kernel_size=cfg.pitch_embedding_kernel_size,
+            padding=int((cfg.pitch_embedding_kernel_size - 1) / 2),
+        )
+
+        # Store values precomputed from training data for convenience
+        self.register_buffer('pitch_mean', torch.zeros(1))
+        self.register_buffer('pitch_std', torch.zeros(1))
+
+        self.loss = BaseFastPitchLoss()
+
+        self.mel_loss_coeff = cfg.mel_loss_coeff
+
+        self.log_train_images = False
+        self.logged_real_samples = False
+        self._tb_logger = None
+        self.hann_window = None
+        self.splice_length = self._cfg.splice_length
+        typecheck.set_typecheck_enabled(enabled=False)  # TODO
+
+    @property
+    def tb_logger(self):
+        if self._tb_logger is None:
+            if self.logger is None and self.logger.experiment is None:
+                return None
+            tb_logger = self.logger.experiment
+            if isinstance(self.logger, LoggerCollection):
+                for logger in self.logger:
+                    if isinstance(logger, TensorBoardLogger):
+                        tb_logger = logger.experiment
+                        break
+            self._tb_logger = tb_logger
+        return self._tb_logger
+
+    @property
+    def parser(self):
+        if self._parser is not None:
+            return self._parser
+
+        self._parser = parsers.make_parser(
+            labels=self._cfg.labels,
+            name='en',
+            unk_id=-1,
+            blank_id=-1,
+            do_normalize=True,
+            abbreviation_version="fastpitch",
+            make_table=False,
+        )
+        return self._parser
+
+    def parse(self, str_input: str) -> torch.tensor:
```

Comment:
Update parse to match latest FS2 parse function?

Reply:
This is the fastpitch model. I'll add parse to the FS2 model